From 28edb057c987bbe3dedbb3376ae4c89568626e8c Mon Sep 17 00:00:00 2001
From: Vince Chang <vince_chang@aspeedtech.com>
Date: Fri, 27 Jun 2025 16:19:47 -0600
Subject: [PATCH] support UFS storage.

- Map UFS storage devices:
  /dev/mmcblk0      => /dev/sda
  /dev/mmcblk0boot0 => /dev/sdb
  /dev/mmcblk0boot1 => /dev/sdc

- Disable force_ro for UFS updates:
  Commented out force_ro handling in mmc_update and mmc_mirroruboot.

Upstream-Status: Pending

Signed-off-by: Vince Chang <vince_chang@aspeedtech.com>
---
 bmc/obmc-flash-bmc | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/bmc/obmc-flash-bmc b/bmc/obmc-flash-bmc
index f9cc4e2..2a47c7f 100644
--- a/bmc/obmc-flash-bmc
+++ b/bmc/obmc-flash-bmc
@@ -576,13 +576,13 @@ function mmc_mount() {
 
 function mmc_update() {
     # Update u-boot if needed
-    bootPartition="mmcblk0boot0"
+    bootPartition="sdb"
     devUBoot="/dev/${bootPartition}"
     imgUBoot="${imgpath}/${version}/image-u-boot"
     if [ "$(cmp_uboot "${devUBoot}" "${imgUBoot}")" != "0" ]; then
-        echo 0 > "/sys/block/${bootPartition}/force_ro"
+        #echo 0 > "/sys/block/${bootPartition}/force_ro"
         dd if="${imgUBoot}" of="${devUBoot}"
-        echo 1 > "/sys/block/${bootPartition}/force_ro"
+        #echo 1 > "/sys/block/${bootPartition}/force_ro"
     fi
 
     # Update the secondary (non-running) boot and rofs partitions.
@@ -592,13 +592,13 @@ function mmc_update() {
     # by getting the partition number mmcblk0pX from their label.
     zstd -d -c "${imgpath}"/"${version}"/image-kernel | dd of="/dev/disk/by-partlabel/boot-${label}"
     number="$(readlink -f /dev/disk/by-partlabel/boot-"${label}")"
-    number="${number##*mmcblk0p}"
-    sgdisk --change-name="${number}":boot-"${label}" /dev/mmcblk0 1>/dev/null
+    number="${number##*sda}"
+    sgdisk --change-name="${number}":boot-"${label}" /dev/sda 1>/dev/null
 
     zstd -d -c "${imgpath}"/"${version}"/image-rofs | dd of="/dev/disk/by-partlabel/rofs-${label}"
     number="$(readlink -f /dev/disk/by-partlabel/rofs-"${label}")"
-    number="${number##*mmcblk0p}"
-    sgdisk --change-name="${number}":rofs-"${label}" /dev/mmcblk0 1>/dev/null
+    number="${number##*sda}"
+    sgdisk --change-name="${number}":rofs-"${label}" /dev/sda 1>/dev/null
 
     # Run this after sgdisk for labels to take effect.
     partprobe
@@ -649,11 +649,11 @@ function mmc_mirroruboot() {
     # Get current boot device; 0-primary_bootdev device; 1 - alt_bootdev
     bootdev=$(cat /sys/kernel/debug/aspeed/sbc/abr_image)
     if [[ "${bootdev}" == "0" ]]; then
-        bootPartition="mmcblk0boot0"
-        alt_bootPartition="mmcblk0boot1"
+        bootPartition="sdb"
+        alt_bootPartition="sdc"
     else
-        bootPartition="mmcblk0boot1"
-        alt_bootPartition="mmcblk0boot0"
+        bootPartition="sdc"
+        alt_bootPartition="sdb"
     fi
 
     devUBoot="/dev/${bootPartition}"
@@ -666,9 +666,9 @@ function mmc_mirroruboot() {
 
     if [[ "${checksum_UBoot}" != "${checksum_alt_UBoot}" ]]; then
         echo "Mirroring U-boot to alt chip"
-        echo 0 > "/sys/block/${alt_bootPartition}/force_ro"
+        #echo 0 > "/sys/block/${alt_bootPartition}/force_ro"
         dd if="${devUBoot}" of="${alt_devUBoot}"
-        echo 1 > "/sys/block/${alt_bootPartition}/force_ro"
+        #echo 1 > "/sys/block/${alt_bootPartition}/force_ro"
     fi
 }
 
-- 
2.43.0

